### PR TITLE
Mark Test Utilities As Packable Nonshipping

### PR DESCRIPTION
**Pull request by lonitra**
Mark some of our test utilities in a project that is packable nonshipping in preparation to share this with WPF. In this PR I've:
- Moved winforms specific test utilities out to the respective test utility assembly they were being used in which for the most part was System.Windows.Forms.TestUtilities
- Renamed System.Windows.Forms.Common.TestUtilities to System.Private.Windows.Core.TestUtilities to match our already shared assembly System.Private.Windows.Core
- Added/updated documentation for remaining classes/methods in System.Private.Windows.Core.TestUtilities

Next step here is to investigate if it is possible to have documentation for classes/methods in our shared assembly show up in intellisense within WPF repo but keep it out of windowsdesktop. Currently developers need to navigate via F12 to see any documentation for System.Private.Windows.Core.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12407)

### CODE REVIEW

File: src/Common/tests/TestUtilities/System.Private.Windows.Core.TestUtilities.csproj [hunk 1]
```diff
@@ -1,9 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <AssemblyName>System.Windows.Forms.Common.TestUtilities</AssemblyName>
+    <AssemblyName>System.Private.Windows.Core.TestUtilities</AssemblyName>
```
[the remark this answers is not in the document]
Comment (lonitra) [reply src/Common/tests/TestUtilities/System.Private.Windows.Core.TestUtilities.csproj:3]: Right now this assembly lives under the test folder. I was not planning to rename it as I think it should still be there.

File: src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using System.Drawing;
 using System.Runtime.InteropServices;
 using Moq;
+using System.Private.Windows.Core.TestUtilities;
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/tests/UnitTests/TextBoxBaseTests.cs:8]: Consider adding the test namespaces to the globals file

File: src/Common/tests/TestUtilities/System.Private.Windows.Core.TestUtilities.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tanya-Solyanik) [reply src/Common/tests/TestUtilities/System.Private.Windows.Core.TestUtilities.csproj:3]: I meant the `common` folder

**Review by Tanya-Solyanik — APPROVED** [review]

File: src/Common/tests/TestUtilities/System.Private.Windows.Core.TestUtilities.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (lonitra) [reply src/Common/tests/TestUtilities/System.Private.Windows.Core.TestUtilities.csproj:3]: Ah I see. I think the folder name is still applicable, but I'm open to renaming.

**Review by JeremyKuhne — APPROVED** [review]

**Pull request merged**